lib/dao.js

```diff
@@ -2638,8 +2638,12 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
   var connector = this.getConnector();
-  assert(typeof connector.replaceById === 'function',
-          'replaceById() must be implemented by the connector');
+
+  if (typeof connector.replaceById !== 'function') {
+    var err =  new Error('there is no support for replaceById() in the connector.' +
+      'Connector needs to add support for `replaceById()`');
+    return cb(err);
+  }
```

Comment:
I do not understand why we are doing this differently from any other connector function (ie. https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L1871). This seems like a breaking change since we won't be throwing errors to stop app execution anymore. I am not confident this change is necessary, @bajtos Can you chime in here?